static/src/js/components/DraftList/DraftList.jsx

```diff
@@ -59,63 +71,181 @@ const DraftList = ({ draftType }) => {
-                new Date(revisionDate).toISOString().split('T')[0]
-              )
-            },
-            {
-              value:
-              (
-                <div className="d-flex">
-                  <Button
-                    className="d-flex"
-                    Icon={FaFileDownload}
-                    onClick={() => handleDownloadClick(conceptId)}
-                    variant="secondary"
-                    size="sm"
-                  >
-                    Download JSON
-                  </Button>
-                </div>
-              )
-            }
-          ]
-      }
+      <EllipsisLink to={`/drafts/${`${paramDraftType.toLowerCase()}`}/${conceptId}`}>
+        {cellData}
+      </EllipsisLink>
     )
-  })
-  )
+  }, [])
+
+  const buildEllipsisTextCell = useCallback((originalCellData) => {
+    let cellData = originalCellData
+
+    if (!cellData && draftType === conceptIdTypes.C) cellData = '<Blank Entry Title>'
+    if (!cellData) cellData = '<Blank Long Name>'
+
+    return (
+      <EllipsisText>
+        {cellData}
+      </EllipsisText>
+    )
+  }, [])
+
+  const buildActionsCell = useCallback((cellData, rowData) => {
+    const { conceptId } = rowData
+
+    return (
+      <div className="d-flex">
+        <Button
+          className="d-flex"
+          Icon={FaFileDownload}
+          iconTitle="Document with an arrow pointing down"
+          onClick={() => handleDownloadClick(conceptId)}
+          variant="secondary"
+          size="sm"
+        >
+          Download JSON
+        </Button>
+      </div>
+    )
+  }, [])
+
+  const [columns, setColumns] = useState([
+    {
+      dataKey: 'revisionDate',
+      title: 'Last Modified',
+      className: 'col-auto',
+      dataAccessorFn: (cellData) => cellData.split('T')[0]
+    },
+    {
+      title: 'Actions',
+      className: 'col-auto',
+      dataAccessorFn: buildActionsCell
+    }
+  ])
+
+  useEffect(() => {
+    let newColumns = [...columns]
+
+    if (draftType === conceptIdTypes.C) {
+      newColumns = [
+        {
+          dataKey: 'ummMetadata.ShortName',
+          title: 'Short Name',
+          className: 'col-auto',
+          dataAccessorFn: buildPrimaryEllipsisLink
+        },
+        {
+          dataKey: 'ummMetadata.EntryTitle',
+          title: 'Entry Title',
+          className: 'col-auto',
+          dataAccessorFn: buildEllipsisTextCell
+        },
+        ...newColumns
+      ]
+    } else {
+      newColumns = [
+        {
+          dataKey: 'ummMetadata.Name',
+          title: 'Name',
+          className: 'col-auto',
+          dataAccessorFn: buildPrimaryEllipsisLink
+        },
+        {
+          dataKey: 'ummMetadata.LongName',
+          title: 'Long Name',
+          className: 'col-auto',
+          dataAccessorFn: buildEllipsisTextCell
+        },
+        ...newColumns
+      ]
+    }
+
+    setColumns(newColumns)
+  }, [])
+
+  const currentPageIndex = Math.floor(offset / limit)
```

Comment:
I'm thinking this code will be repeated in multiple places, consider moving into a helper function where you just give it the offset, limit, and active page and will will return all these variables.  I see it is repeated in other places